lib/login_page.dart

```diff
@@ -45,26 +63,115 @@ class _LoginPageState extends State<LoginPage> {
     });
   }
 
+  void signIn(_formKey) {
+    if (_formKey.currentState.validate()) {
+      Navigator.push(
+        context,
+        MaterialPageRoute(
+          builder: (context) {
+            return HomePage();
+          },
+        ),
+      );
+    }
+  }
+
+  @override
+  Future<void> resetPassword(String email) async {
+    await _auth.sendPasswordResetEmail(email: email);
+  }
+
+  Widget _form(BuildContext context) {
+    RegExp regExp;
+    return Form(
+        key: _formKey,
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: <Widget>[
+            TextFormField(
+              keyboardType: TextInputType.emailAddress,
+              decoration: const InputDecoration(
+                hintText: 'Email',
+              ),
+              validator: (value) {
+                if (value.isEmpty) {
+                  return 'Email needed';
+                } else if (!_emailValid.hasMatch(value)) {
+                  return 'Invalid Input';
+                }
+                return null;
+              },
+              //controller: _emailController,
+              textAlign: TextAlign.center,
+            ),
+            TextFormField(
+              obscureText: true,
+              decoration: const InputDecoration(
+                hintText: 'Password',
+              ),
+              validator: (value) {
+                if (value.isEmpty) {
+                  return 'Password Needed';
+                } else if (_passwordValid.hasMatch(value)) {
+                  return 'Missing something';
+                }
+                return null;
+              },
+              controller: _passwordController,
+              textAlign: TextAlign.center,
+            ),
+            Padding(
+              padding: const EdgeInsets.symmetric(vertical: 16.0),
+              child: ElevatedButton(
+                onPressed: () {
+                  signIn(_formKey);
+                },
+                child: Text('Submit'),
+              ),
+            ),
+            Padding(
+                padding: const EdgeInsets.symmetric(vertical: 16.0),
+                child: ElevatedButton(
+                  onPressed: () async {
+                    Navigator.push(
+                      context,
+                      MaterialPageRoute(
+                        builder: (context) {
+                          return SignupPage();
+                        },
+                      ),
+                    );
+                  },
+                  child: Text('signup'),
+                )),
+          ],
+        ));
+  }
+
+  Widget _forgotPassword(BuildContext context) {
+    return TextButton(onPressed: null, child: Text('forgot password?'));
+  }
+
```

Comment:
This button doesn't work, find a way to redirect the user to another screen so they can reset their password